packages/circuits/circuits/epochKeyLite.circom

```diff
@@ -18,21 +18,27 @@ template EpochKeyLite(EPOCH_KEY_NONCE_PER_EPOCH) {
      * 1 bit reveal nonce
      **/
 
-    control \ (2 ** 233) === 0;
+    signal control_divided <-- control \ (2 ** 233);
```

Comment:
I think this approach to checking the range of values doesn't work. I thought we could apply a constraint to division by a constant but this was apparently a bug that they fixed in 2.1.3. This division has to be constrained because if the value is > 2^233 then the check at line 37 can be broken.

Now we need to use a `LessThan` component for each of these checks (and convert each value to bits with `Num2Bits`). We actually need to use `Num2Bits_strict` because the input comes from the user/unconstrained. I think it's worth building this and seeing how many constraints it adds.

Another approach is accepting the `control` signal as `control[233]` e.g. a binary array. Then we can iterate over each value and check that (`control[x] == 0 || control[x] == 1`). Then take the relevant sections and convert them to numbers.

^ This should add the fewest constraints but requires us to convert the number to a binary array in javascript.